include/cuco/detail/static_map.inl

```diff
@@ -22,36 +22,42 @@ template <typename Key, typename Value, cuda::thread_scope Scope, typename Alloc
 static_map<Key, Value, Scope, Allocator>::static_map(std::size_t capacity,
                                                      Key empty_key_sentinel,
                                                      Value empty_value_sentinel,
-                                                     Allocator const& alloc)
+                                                     Allocator const& alloc,
+                                                     cudaStream_t stream)
   : capacity_{std::max(capacity, std::size_t{1})},  // to avoid dereferencing a nullptr (Issue #72)
     empty_key_sentinel_{empty_key_sentinel},
     empty_value_sentinel_{empty_value_sentinel},
     slot_allocator_{alloc},
-    counter_allocator_{alloc}
+    counter_allocator_{alloc},
+    exec_stream_{stream}
 {
-  slots_         = std::allocator_traits<slot_allocator_type>::allocate(slot_allocator_, capacity_);
-  num_successes_ = std::allocator_traits<counter_allocator_type>::allocate(counter_allocator_, 1);
+  // allocator should allocate memory accessible by the exec_stream_
+  slots_         = slot_allocator_.allocate(capacity_, exec_stream_);
+  num_successes_ = counter_allocator_.allocate(1, exec_stream_);
```

Comment:
Hm, this is going to break users that weren't passing an Allocator that takes a stream.

Reply:
Since `cuco::allocator::allocate` is also not doing anything with stream currently, should we rather just remove the stream associated memory allocation for now?
Doesn't seem like there is an existing cuco deployment where memory was associated with a specific stream. Future work can account for a std::allocator interface with stream attachment?

Reply:
I was thinking about this issue but haven't figured out a good solution. https://github.com/rapidsai/cudf/pull/9666 depends on this PR and it requires stream argument to work properly with rmm memory pool.

Reply:
For https://github.com/rapidsai/cudf/pull/9666 we can use https://github.com/rapidsai/rmm/blob/bbfb0119a2e70ece729628810976f0e03e9935d9/include/rmm/mr/device/polymorphic_allocator.hpp#L126-L149

Reply:
Right, so we can remove the stream argument in `cuco::allocator` and do not pass stream to `allocator::allocate` function in both single map and multimap.

Reply:
Made a PR to remove the stream argument in `cuco::allocator`, @chirayuG-nvidia Can you please remove the stream in both allocate and deallocate here? Also, since stream is no longer required in the destructor. We can get rid of the `exec_stream_` member variable as well.

Reply:
Done.